iotdb-client/session/src/main/java/org/apache/iotdb/session/pool/TableSessionWrapper.java

```diff
@@ -131,67 +107,19 @@ public void executeNonQueryStatement(String sql)
   }
 
   @Override
-  public String getTimestampPrecision() throws TException {
-    try {
-      return session.getTimestampPrecision();
-    } catch (TException e) {
-      sessionPool.cleanSessionAndMayThrowConnectionException(session);
-      closed.set(true);
-      session = null;
-      throw e;
-    }
-  }
-
-  @Override
-  public void insertTablet(Tablet tablet)
-      throws StatementExecutionException, IoTDBConnectionException {
-    try {
-      session.insertRelationalTablet(tablet);
-    } catch (IoTDBConnectionException e) {
-      sessionPool.cleanSessionAndMayThrowConnectionException(session);
-      closed.set(true);
-      session = null;
-      throw e;
-    }
-  }
-
-  @Override
-  public boolean isEnableQueryRedirection() {
-    return session.isEnableQueryRedirection();
-  }
-
-  @Override
-  public boolean isEnableRedirection() {
-    return session.isEnableRedirection();
-  }
-
-  @Override
-  public TSBackupConfigurationResp getBackupConfiguration()
-      throws IoTDBConnectionException, StatementExecutionException {
-    try {
-      return session.getBackupConfiguration();
-    } catch (IoTDBConnectionException e) {
-      sessionPool.cleanSessionAndMayThrowConnectionException(session);
-      closed.set(true);
-      session = null;
-      throw e;
-    }
-  }
-
-  @Override
-  public TSConnectionInfoResp fetchAllConnections() throws IoTDBConnectionException {
-    try {
-      return session.fetchAllConnections();
-    } catch (IoTDBConnectionException e) {
-      sessionPool.cleanSessionAndMayThrowConnectionException(session);
-      closed.set(true);
+  public void close() throws IoTDBConnectionException {
+    if (closed.compareAndSet(false, true)) {
+      if (!Objects.equals(session.getDatabase(), sessionPool.database)
+          && sessionPool.database != null) {
+        try {
+          session.executeNonQueryStatement("use " + sessionPool.database);
+        } catch (StatementExecutionException e) {
+          LOGGER.warn(
+              "Failed to change back database by executing: use {}", sessionPool.database, e);
+        }
+      }
+      sessionPool.putBack(session);
```

Comment:
If the execution fails, is it still necessary to put the session back?

Reply:
OK, fixed.